pkg/run/session/connect/connect.go

```diff
@@ -607,8 +607,7 @@ func (conn *Connection) createServiceAccountToken(vKubeConfig api.Config) (strin
 	token := ""
 	conn.Log.Actionf("Create service account token for %s/%s", serviceAccountNamespace, serviceAccount)
 
-	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
-	err = wait.Poll(time.Second, time.Minute*3, func() (bool, error) {
+	err = wait.PollUntilContextTimeout(context.Background(), time.Second*2, time.Minute*5, true, func(_ context.Context) (bool, error) {
```

Comment:
```suggestion
	err = wait.PollUntilContextTimeout(context.Background(), time.Second, time.Minute*3, true, func(_ context.Context) (bool, error) {
```
I'm assuming this needs to stay the same as before, if not please ignore this comment.

Reply:
Yes, that happens when I use vim macros.